clang/lib/Sema/SemaCoroutine.cpp

```diff
@@ -684,6 +684,18 @@ bool Sema::checkFinalSuspendNoThrow(const Stmt *FinalSuspend) {
   return ThrowingDecls.empty();
 }
 
+// [stmt.return.coroutine]p1:
+//   A coroutine shall not enclose a return statement ([stmt.return]).
+static void checkReturnStmtInCoroutine(Sema &S, FunctionScopeInfo *FSI) {
+  if (FSI && FSI->FirstReturnLoc.isValid()) {
+      assert(FSI->FirstCoroutineStmtLoc.isValid() &&
```

Comment:
NIT: we could move this assertion to the start of the function. As the name implies, the `checkReturnStmtInCoroutine` can only be called on coroutines, so validating this invariant is useful even if we don't report any diagnostics.

Reply:
Done.